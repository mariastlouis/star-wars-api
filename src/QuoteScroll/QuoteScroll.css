@import url("https://fonts.googleapis.com/css?family=Pathway+Gothic+One");
@import url("https://fonts.googleapis.com/css?family=Cabin");
body {
  background: black;
  color: white;
  font-family: "Cabin", sans-serif; }

h1 {
  font-size: 20px; }

h2 {
  display: block;
  -webkit-margin-before: 0;
  -webkit-margin-after: 0;
  -webkit-margin-start: 0px;
  -webkit-margin-end: 0px; }

.quoteScroll {
  width: 100%;
  height: 0;
  overflow: hidden;
  animation: addHeight 16s linear; }

.fade {
  position: relative;
  width: 100%;
  min-height: 20vh;
  top: -25px;
  z-index: 1; }

.star-wars-quote {
  display: flex;
  justify-content: center;
  position: relative;
  height: 150px;
  color: #feda4a;
  font-family: "Pathway Gothic One", sans-serif;
  font-size: 300%;
  font-weight: 600;
  letter-spacing: 6px;
  line-height: 150%;
  perspective: 400px;
  text-align: justify; }

.crawl {
  position: relative;
  top: 99999px;
  transform-origin: 50% 100%;
  animation: crawl 60s linear; }

.crawl > .title {
  font-size: 90%;
  text-align: center; }

.crawl > .title h1 {
  margin: 0 0 100px;
  text-transform: uppercase;
  font-size: 50px; }

@keyframes crawl {
  0% {
    top: -100px;
    transform: rotateX(20deg) translateZ(0); }
  100% {
    top: -6000px;
    transform: rotateX(25deg) translateZ(-2500px); } }

@keyframes addHeight {
  0% {
    height: 200px; }
  100% {
    height: 200px; } }
