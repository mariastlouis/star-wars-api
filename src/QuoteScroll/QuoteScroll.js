import React from 'react'
import PropTypes from 'prop-types';


const QuoteScroll = ({movie}) => {
  return (
    <div className = "quoteScroll">
     {movie.openingCrawl}
     {movie.title}

    </div>
  )
}

export default QuoteScroll;