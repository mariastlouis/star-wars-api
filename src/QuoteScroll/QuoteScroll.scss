@import 'variable';

.quoteScroll {
  width: 100%;
  height: 100%;
  overflow: hidden;
}

.fade {
  position: relative;
  width: 100%;
  min-height: 20vh;
  top: -25px;
  z-index: 1;
}

.star-wars-quote {
  display: flex;
  justify-content: center;
  position: relative;
  height: 150px;
  color: $accent-color;
  font-family: $accent-font;
  font-size: 300%;
  font-weight: 600;
  letter-spacing: 6px;
  line-height: 150%;
  perspective: 400px;
  text-align: justify;
}

.crawl {
  position: relative;
  top: 99999px;
  transform-origin: 50% 100%;
  animation: crawl 60s linear;
}

.crawl > .title {
  font-size: 90%;
  text-align: center;
}

.crawl > .title h1 {
  margin: 0 0 100px;
  text-transform: uppercase;
  font-size: 50px;
}

@keyframes crawl {
    0% {
    top: -100px;
    transform: rotateX(20deg)  translateZ(0);
  }
    100% { 
    top: -6000px;
    transform: rotateX(25deg) translateZ(-2500px);
  }
}