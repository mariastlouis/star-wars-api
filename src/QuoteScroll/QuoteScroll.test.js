import React from 'react';
import QuoteScroll from './QuoteScroll.js';
import { shallow } from 'enzyme';

const mockRandom = jest.fn();

const mockFilm = {
  openingCrawl: "It is a period of civil war.",
  releaseDate: "1977-05-25",
  title: "A New Hope"
}

let renderedQuote;

describe('QuoteScroll' , () =>{
  
  beforeEach(() =>{
    renderedQuote = shallow(<QuoteScroll movie = {mockFilm} />)
  });

  it('should render correctly', () =>{
    expect(renderedQuote).toBeDefined();
  });

  it('should have classes of title and crawl', () =>{
    expect(renderedQuote.find('.crawl').length).toEqual(1);
    expect(renderedQuote.find('.title').length).toEqual(1);
  });

  it('should match the snapshot', () =>{
    expect(renderedQuote).toMatchSnapshot();
  });

});
