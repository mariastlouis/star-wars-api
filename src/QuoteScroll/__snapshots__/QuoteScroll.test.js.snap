// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuoteScroll should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <QuoteScroll
    movie={
        Object {
            "openingCrawl": "It is a period of civil war.",
            "releaseDate": "1977-05-25",
            "title": "A New Hope",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="fade"
/>,
        <div
          className="star-wars-quote"
>
          <div
                    className="crawl"
          >
                    <div
                              className="title"
                    >
                              <h1>
                                        A New Hope
                              </h1>
                    </div>
                    It is a period of civil war.
          </div>
</div>,
      ],
      "className": "quoteScroll",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "fade",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="crawl"
>
            <div
                        className="title"
            >
                        <h1>
                                    A New Hope
                        </h1>
            </div>
            It is a period of civil war.
</div>,
          "className": "star-wars-quote",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="title"
>
                <h1>
                                A New Hope
                </h1>
</div>,
              "It is a period of civil war.",
            ],
            "className": "crawl",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <h1>
                  A New Hope
</h1>,
                "className": "title",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "A New Hope",
                },
                "ref": null,
                "rendered": "A New Hope",
                "type": "h1",
              },
              "type": "div",
            },
            "It is a period of civil war.",
          ],
          "type": "div",
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="fade"
/>,
          <div
            className="star-wars-quote"
>
            <div
                        className="crawl"
            >
                        <div
                                    className="title"
                        >
                                    <h1>
                                                A New Hope
                                    </h1>
                        </div>
                        It is a period of civil war.
            </div>
</div>,
        ],
        "className": "quoteScroll",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "fade",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="crawl"
>
              <div
                            className="title"
              >
                            <h1>
                                          A New Hope
                            </h1>
              </div>
              It is a period of civil war.
</div>,
            "className": "star-wars-quote",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="title"
>
                  <h1>
                                    A New Hope
                  </h1>
</div>,
                "It is a period of civil war.",
              ],
              "className": "crawl",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <h1>
                    A New Hope
</h1>,
                  "className": "title",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "A New Hope",
                  },
                  "ref": null,
                  "rendered": "A New Hope",
                  "type": "h1",
                },
                "type": "div",
              },
              "It is a period of civil war.",
            ],
            "type": "div",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
